hamilton/lifecycle/default.py

```diff
@@ -584,10 +598,43 @@ def run_to_execute_node(
         # and truncate it/provide sentinels for every failure)
         # TODO -- decide what to do with collect
         """Executes a node. If the node fails, returns the sentinel value."""
-        for key, value in node_kwargs.items():
-            if value == self.sentinel_value:  # == versus is
-                return self.sentinel_value  # cascade it through
+        default_return = [self.sentinel_value] if is_expand else self.sentinel_value
+        _node_tags = future_kwargs["node_tags"]
+        can_inject = any(
+            _node_tags.get(k, "") == v for k, v in self.sentinel_injection_tags.items()
+        )
+        if not can_inject:
+            for key, value in node_kwargs.items():
+                if type(self.sentinel_value) is type(value):
+                    if self.sentinel_value == value:  # == versus is
+                        return default_return
+        if not is_expand:
+            try:
+                return node_callable(**node_kwargs)
+            except self.error_to_catch:
+                return self.sentinel_value
+
+        # Grab the partial-ized function that is a parallelizable.
+        # Be very specific...
+        if len(node_callable.keywords) == 1 and "_callable" in node_callable.keywords:
+            gen_func = node_callable.keywords["_callable"]
+        elif self.fail_all_parallel:
+            gen_func = node_callable
+        else:
+            raise ValueError(
+                "Unexpected configuration for expandable node and GracefulErrorAdapter."
+            )
         try:
-            return node_callable(**node_kwargs)
+            gen = gen_func(**node_kwargs)
         except self.error_to_catch:
-            return self.sentinel_value
+            return [self.sentinel_value]
+        results: list[Any] = []
+        try:
+            for _res in gen:
+                results.append(_res)
+        except self.error_to_catch:
+            if self.fail_all_parallel:
+                results = [self.sentinel_value]
+            else:
+                results.append(self.sentinel_value)
```

Comment:
Just to check, if item (3) fails out of 1,2,3,4,5, 4/5 will still be run? Might be worth adding in the docstring.

Reply:
4 and 5 will not be run.

Updated the docstring with an example:

[f7c0e94](https://github.com/DAGWorks-Inc/hamilton/pull/1017/commits/f7c0e94de67aa969574e25ca2919768a534b8b31)